modules/mcuboot/CMakeLists.txt

```diff
@@ -6,14 +6,33 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
 
   include(${ZEPHYR_BASE}/../nrf/cmake/fw_zip.cmake)
 
-  function(sign to_sign_hex output_prefix offset sign_depends signed_hex_out)
-    set(op ${output_prefix})
-    set(signed_hex ${op}_signed.hex)
-    set(${signed_hex_out} ${signed_hex} PARENT_SCOPE)
-    set(to_sign_bin ${op}_to_sign.bin)
-    set(update_bin ${op}_update.bin)
-    set(moved_test_update_hex ${op}_moved_test_update.hex)
-    set(test_update_hex ${op}_test_update.hex)
+  function(sign)
+    # Signs a hex image
+    #
+    # Required arguments are:
+    # HEX_TO_SIGN - Hex image to sign
+    # SIGNED_PREFIX - Prefix of the output image
+    # OFFSET - offset
+    # HEX_SIGNED - Signed hex output image
+    # DEPENDS - One or more dependencies for signing the hex image
+
+    set(oneValueArgs HEX_TO_SIGN SIGNED_PREFIX OFFSET HEX_SIGNED)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(SIGN_ARG "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if (NOT SIGN_ARG_HEX_TO_SIGN
```

Comment:
Better to use `NOT DEFINED` here and below.
Otherwise a call such as:
```
sign(... SIGN_ARG_SIGNED_PREFIX false ...)
```
will fail, as `NOT SIGN_ARG_SIGNED_PREFIX` will be true, even though SIGN_ARG_SIGNED_PREFIX is defined.

Reply:
Seems you didn't fix my comment.
> Better to use NOT DEFINED here and below.


you still have several `NOT <varirable>` instead of `NOT DEFINED <variable>` here:
https://github.com/nrfconnect/sdk-nrf/blob/fbf89b2f5cfb43dd59d961d9f000c964870706d2/modules/mcuboot/CMakeLists.txt#L23-L26

Reply:
Oops, overlooked these. beyond the first line. Fixed now.